servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/Flush.java

```diff
@@ -74,24 +73,22 @@ private Flush() {
         public void onSubscribe(Subscription subscription) {
             try {
                 writeEventsListener.writeStarted();
-            } catch (Throwable t) {
-                subscription.cancel();
-                subscriber.onSubscribe(EMPTY_SUBSCRIPTION);
-                subscriber.onError(t);
-                return;
-            }
-            subscriber.onSubscribe(new Subscription() {
-                @Override
-                public void request(long n) {
-                    subscription.request(n);
-                }
+            } finally {
+                // As long as we call onSubscribe we can let exceptions propagate and the source should terminate
+                // this Subscriber for cleanup.
+                subscriber.onSubscribe(new Subscription() {
```

Comment:
This seems to be different than what was happening before as we are not cancelling the `subscription` any more. I think the assumption here is that this exception will bubble up to the source which effectively cancels the `subscription`?

Reply:
Correct. I attempted to articulate that in the comment above but let me expand on this to clarify more.